submissions/AnnaStoyano/oop-exercise/index.js

```diff
@@ -0,0 +1,112 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: _put repo URL here_
+   Web app: _put project's github pages URL here_
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+
+class Mammals{
+   constructor(name,gender,friends){
+      this.species = 'mammals',
+      this.name = name,
+      this.gender = gender,
+      this.legs = 4,
+      this.friends = friends
+   }
+}
+
+class Dog extends Mammals{
+   constructor(name,gender,friends,saying){
+      super(name,gender,friends);
+      this.species = 'dog',
+      this.saying = saying || 'guf'
+   }
+};
+
+class Cat extends Mammals{
+   constructor(name,gender,friends,saying){
+      super(name,gender,saying,friends);
+      this.species = 'cat',
+      this.saying = saying || 'guf'
+   }
+
+   getSaying(){
+      return this.saying || 'meow';
+   }
+}
+
+class Human extends Mammals{
+   constructor(name,gender,friends,saying){
+      super(name,gender,friends);
+      this.species = 'human'
+      this.hands = 2;
+      this.legs =2;
+      this.saying = saying || `Hello! My name is ${this.name}`
+   }
+}
+
+class Woman extends Human{
+   constructor(name,friends,saying){
+      super(name,friends,saying);
+      this.gender = 'female'
+   }
+}
+
+class Man extends Human{
+   constructor(name,friends,saying){
+      super(name,friends,saying);
+      this.gender = 'male'
+   }
+}
+
+class CatWoman extends Woman{
+   constructor(name,friends,saying){
+      super(name,friends,saying);
+      this.saying = Cat.prototype.getSaying();
+   }
+}
+
+const cat = new Cat(name='Murzic',gender='male',friends=['Elza','Nicky'])
+const dog = new Dog(name='Lucy',gender='female',saying='guffyy',friends=['Alla','Dima']);
+const man = new Man(name='Vlad',friends=['Sergay','Anton'],saying='Hello everyone!');
+const woman = new Woman(name='Anna',friends=['Sergay','Artem']);
+const catWoman = new CatWoman(name='Lily',friends=['Batman'])
+
+const inhabitants = [man,woman,cat,dog,catWoman];
+
+const properties = ['name','species','gender','legs','hands','saying','friends'];
+let output = inhabitants.map(inhabitant=>
+   properties.map(property=>{
```

Comment:
Let's make construction of a self-presentation string a part of classes.
This way you will avoid `.hasOwnProperty` check as every class has very few properties so it handle those properly.

You will also benefit from OOP concepts of inheritance and methods overloading.

Hints:
Each class should handle only properties know to it.
`super.method()` calls a `method` from its super class.